pkg/exchange/bitget/convert.go

```diff
@@ -59,3 +63,209 @@ func toGlobalTicker(ticker bitgetapi.Ticker) types.Ticker {
+	}
+
+	return &types.Order{
+		SubmitOrder: types.SubmitOrder{
+			ClientOrderID: order.ClientOrderId,
+			Symbol:        order.Symbol,
+			Side:          side,
+			Type:          orderType,
+			Quantity:      qty,
+			Price:         price,
+			// Bitget does not include the "time-in-force" field in its API response for spot trading, so we set GTC.
+			TimeInForce: types.TimeInForceGTC,
+		},
+		Exchange:         types.ExchangeBitget,
+		OrderID:          uint64(order.OrderId),
+		UUID:             strconv.FormatInt(int64(order.OrderId), 10),
+		Status:           status,
+		ExecutedQuantity: order.BaseVolume,
+		IsWorking:        order.Status.IsWorking(),
+		CreationTime:     types.Time(order.CTime.Time()),
+		UpdateTime:       types.Time(order.UTime.Time()),
+	}, nil
+}
+
+func toGlobalOrder(order v2.OrderDetail) (*types.Order, error) {
+	side, err := toGlobalSideType(order.Side)
+	if err != nil {
+		return nil, err
+	}
+
+	orderType, err := toGlobalOrderType(order.OrderType)
+	if err != nil {
+		return nil, err
+	}
+
+	status, err := toGlobalOrderStatus(order.Status)
+	if err != nil {
+		return nil, err
+	}
+
+	qty := order.Size
+	price := order.Price
+
+	if orderType == types.OrderTypeMarket {
+		price = order.PriceAvg
+		if side == types.SideTypeBuy {
+			qty, err = processMarketBuyQuantity(order.BaseVolume, order.QuoteVolume, order.PriceAvg, order.Size, order.Status)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	return &types.Order{
+		SubmitOrder: types.SubmitOrder{
+			ClientOrderID: order.ClientOrderId,
+			Symbol:        order.Symbol,
+			Side:          side,
+			Type:          orderType,
+			Quantity:      qty,
+			Price:         price,
+			// Bitget does not include the "time-in-force" field in its API response for spot trading, so we set GTC.
+			TimeInForce: types.TimeInForceGTC,
+		},
+		Exchange:         types.ExchangeBitget,
+		OrderID:          uint64(order.OrderId),
+		UUID:             strconv.FormatInt(int64(order.OrderId), 10),
+		Status:           status,
+		ExecutedQuantity: order.BaseVolume,
+		IsWorking:        order.Status.IsWorking(),
+		CreationTime:     types.Time(order.CTime.Time()),
+		UpdateTime:       types.Time(order.UTime.Time()),
+	}, nil
+}
+
+// processMarketBuyQuantity returns the estimated base quantity or real. The order size will be 'quote quantity' when side is buy and
+// type is market, so we need to convert that. This is because the unit of types.Order.Quantity is base coin.
+//
+// If the order status is PartialFilled, return estimated base coin quantity.
+// If the order status is Filled, return the filled base quantity instead of the buy quantity, because a market order on the buy side
+// cannot execute all.
+// Otherwise, return zero.
+func processMarketBuyQuantity(filledQty, filledPrice, priceAvg, buyQty fixedpoint.Value, orderStatus v2.OrderStatus) (fixedpoint.Value, error) {
```

Comment:
bitget uses the same way as bybit?

Reply:
There are some similarities; I think they can be combined later